docs/src/main/docs/guides/07_datasource.adoc

```diff
@@ -0,0 +1,364 @@
+database driver library]:
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.h2database</groupId>
+    <artifactId>h2</artifactId>
+    <version>1.4.199</version>
+    <scope>runtime</scope> <!--1-->
+</dependency>
+----
+<1> Note the `<scope>` is `runtime`.
+
+== Add a Helidon Connection Pool Extension to Your Helidon MP Application's Runtime Classpath
+
+Helidon contains two <<extensions/01_overview.adoc,extensions>> that
+supply _connection pools_, infrastructure that allows efficient use of
+database connections:
+
+1. <<extensions/02_cdi_datasource-hikaricp.adoc,HikariCP extension>>:
+   an extension that integrates the
+   https://github.com/brettwooldridge/HikariCP[Hikari connection pool]
+   into Helidon MP applications
+
+2. <<extensions/02_cdi_datasource-ucp.adoc,Oracle Universal Connection
+   Pool extension>>: an extension that integrates
+   https://docs.oracle.com/en/database/oracle/oracle-database/19/jjucp/index.html[Oracle's
+   Universal Connection Pool] into Helidon MP applications
+
+Each connection pool offers tradeoffs in terms of size, efficiency and
+features.
+
+=== Add the Hikari Connection Pool to the Runtime Classpath
+
+If you choose to integrate the Hikari connection pool into your
+Helidon MP application, then add an appropriate
+https://maven.apache.org/ref/3.6.1/maven-model/maven.html#class_dependency[`<dependency>`]
+element as a child element of the `<dependencies>` element in your
+`pom.xml`, referencing the HikariCP extension:
+
+[source,xml,subs="attributes+"]
+----
+<dependency>
+    <groupId>io.helidon.integrations.cdi</groupId>
+    <artifactId>helidon-integrations-cdi-datasource-hikaricp</artifactId>
+    <scope>runtime</scope> <!--1-->
+</dependency>
+----
+<1> Note the `<scope>` is `runtime`.
+
+=== Add the Oracle Universal Connection Pool to the Runtime Classpath
+
+If, instead, you shoose to integrate the Oracle Universal Connection Pool into your Helidon MP application, then add an appropriate
+https://maven.apache.org/ref/3.6.1/maven-model/maven.html#class_dependency[`<dependency>`]
+element as a child element of the `<dependencies>` element in your
+`pom.xml`, referencing the Oracle Universal Connection Pool extension:
+
+[source,xml,subs="attributes+"]
+----
+<dependency>
+    <groupId>io.helidon.integrations.cdi</groupId>
+    <artifactId>helidon-integrations-cdi-datasource-ucp</artifactId>
+    <scope>runtime</scope> <!--1-->
+</dependency>
+----
+<1> Note the `<scope>` is `runtime`.
+
+== Configure the Connection Pool
+
+Regardless of which connection pool you integrate into your Helidon MP
+application via the mechanisms described above, you will need to tell
+that connection pool's machinery how many pools to set up, what to
+name each of them so you can refer to them inside your code, and how
+each of them should connect to a database.
+
+Helidon MP, because it is a conformant MicroProfile specification
```

Comment:
This paragraph should be part of the Helidon MP config documentation IMO.

Reply:
OK; would you like me to remove it?

Reply:
Let's see what others have to say about this, @barchetta thoughts ?

Reply:
Yes, this could be more concise. Don't need to explain MicroProfile config, just that the extension uses it for configuration, and so we'll be putting properties here. And for more information see blah, blah, blah.

Reply:
Totally fair.  Do note that I'm also trying to explain why I'm about to show you some `application.yaml`, but you could supply the same information as system properties or environment variables, in which case the syntax and names of the properties in question change.  I'm not sure if that's something I need to spell out here or not.